ietf/utils/templatetags/datetime_filters.py

```diff
@@ -0,0 +1,49 @@
+# Copyright The IETF Trust 2020, All Rights Reserved
+# -*- coding: utf-8 -*-
+
+import datetime
+
+from django.template import Library, Node, TemplateSyntaxError
+from django.template.defaultfilters import date
+from django.utils import timezone
+
+import debug                            # pyflakes:ignore
+
+register = Library()
+
+elide_timefmt = str.maketrans("aAefgGhHiIOPsTuZ:", "                 ")
+
+@register.filter()
+def dateformat(value, arg=None):
+    """
+    Formats a date or datetime according to the given format.
+    Ignores the time-related format elements if a date is given.
+    """
+    if value in (None, ''):
+        return ''
+    if   isinstance(value, datetime.datetime):
+        pass
+    elif isinstance(value, datetime.date):
+        arg = arg.translate(elide_timefmt).strip()
+    return date(value, arg)
+
+
+class GetNowNode(Node):
+    """Node that stores timezone.now() in a template variable"""
+    def __init__(self, var_name):
+        self.var_name = var_name
+
+    def render(self, context):
+        context[self.var_name] = timezone.now()
+        return ''  # render nothing
```

Comment:
This works by side-effect of render. Is this common in the Django tree already?

I wonder if a context processor (see ietf.contextprocessors.py) injected via middleware might be better?

Reply:
(Or if there's maybe _already_ something like "when the request arrived" in the context?)

Reply:
That is the method in the documentation for setting context variables (https://docs.djangoproject.com/en/2.2/howto/custom-template-tags/#setting-a-variable-in-the-context). The render method is the hook that's available to do things when rendering the template tag.

I'll see if there's something in the `Request` or otherwise in the context. I have mixed feelings about injecting things into the context - wariness of global variables. It would have the advantage here of preventing skew, for what that's worth.

Reply:
As far as I can tell there's no way to get it from the request. I think it'll work to inject it as `timezone_now` in the context. I don't think we're likely to have conflicts with that.

Reply:
Given the doc you point to, lets leave it as is.